src/main/java/org/wso2/carbon/connector/SplitFile.java

```diff
@@ -0,0 +1,173 @@
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+package org.wso2.carbon.connector;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.commons.io.input.AutoCloseInputStream;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.commons.vfs2.FileObject;
+import org.apache.commons.vfs2.FileSystemException;
+import org.apache.commons.vfs2.FileSystemOptions;
+import org.apache.commons.vfs2.FileType;
+import org.apache.commons.vfs2.impl.StandardFileSystemManager;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseException;
+import org.codehaus.jettison.json.JSONException;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.Connector;
+import org.wso2.carbon.connector.core.util.ConnectorUtils;
+import org.wso2.carbon.connector.util.FileConnectorUtils;
+import org.wso2.carbon.connector.util.FileConstants;
+import org.wso2.carbon.connector.util.ResultPayloadCreate;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * Splits the file into multiple chunks and writes them to a file system.
+ */
+public class SplitFile extends AbstractConnector implements Connector {
+    private static final Log log = LogFactory.getLog(SplitFile.class);
+    private StandardFileSystemManager manager = null;
+
+    public void connect(MessageContext messageContext) {
+        String fileLocation = (String) ConnectorUtils.lookupTemplateParamater(messageContext,
+                FileConstants.FILE_LOCATION);
+        String chunkSize = (String) ConnectorUtils.lookupTemplateParamater(messageContext,
+                FileConstants.CHUNK_SIZE);
+        String destination = (String) ConnectorUtils.lookupTemplateParamater(messageContext,
+                FileConstants.NEW_FILE_LOCATION);
+        FileSystemOptions options = FileConnectorUtils.init(messageContext);
+        boolean resultStatus = splitFile(fileLocation, chunkSize, destination, options, messageContext);
+        generateOutput(messageContext, resultStatus);
+    }
+
+    /**
+     * @param fileLocation Location of the source file.
+     * @param chunkSize size of the chunks to split the file.
+     * @param destination   Location of the destination to write the splitted files.
+     * @param options Init configuration options.
+     * @param messageContext    Message context.
+     * @return Status true/false.
+     */
+    private boolean splitFile(String fileLocation, String chunkSize, String destination, FileSystemOptions options,
```

Comment:
What is the use case we are trying to cater here? When splitting the based on capacity we may loose continuation of data?